file/src/main/scala/akka/stream/alpakka/file/scaladsl/Directory.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.file.scaladsl
+
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.{Files, Path}
+import java.util.function.BiPredicate
+
+import akka.NotUsed
+import akka.stream.scaladsl.{Source, StreamConverters}
+
+object Directory {
+
+  /**
+   * List all files in the given directory
+   */
+  def ls(directory: Path): Source[Path, NotUsed] = {
+    require(Files.isDirectory(directory), s"Path must be a directory, $directory isn't")
+    StreamConverters.fromJavaStream(() => Files.list(directory))
+  }
+
+  /**
+   * Recursively list files in the given directory and its subdirectories. Listing is done
+   * depth first.
+   */
+  def walk(directory: Path): Source[Path, NotUsed] = {
```

Comment:
I think it would be good to expose the overloads too: 
https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#walk-java.nio.file.Path-int-java.nio.file.FileVisitOption...-
the visit options and depth seem very useful when someone would be using this API